common/global_utils.py

```diff
@@ -0,0 +1,235 @@
+            access_key=os.getenv("MINIO_ACCESS_KEY"),
+            secret_key=os.getenv("MINIO_SECRET_KEY"),
+            secure=True,
+        )
+
+    def get_qdrant_database_file(self, file_path: PosixPath, object_name: PosixPath):
+        """
+        Pulls zipped Qdrant database snapshot from cloud's object_name to container's file_path.
+        """
+        self._get_minio_client().fget_object(
+            bucket_name=self.minio_bucket_name,
+            object_name=str(object_name),
+            file_path=str(file_path),
+        )
+
+    def get_best_score_imgs(self, results: list[ScoredPoint]) -> list[Image.Image]:
+        """
+        Handler for returning images with the highest similarity scores from cloud storage.
+        Additionally, filenames are returned as future captions in fornt-end module.
+        """
+        object_list = [self.minio_main_path / r.payload["file"] for r in results]
+        return [
+            Image.open(
+                self._get_minio_client().get_object(
+                    bucket_name=self.minio_bucket_name, object_name=str(obj)
+                )
+            )
+            for obj in object_list
+        ]
+
+    def get_random_images_from_collection(self, collection_name: str, k: int) -> tuple[list[str], list[Image.Image]]:
+        """
+        Pulls a random set of images from a selected collection in cloud storage.
+        Used for image input suggestion in front-end component.
+        Additionally, filenames are returned as captions.
+        """
+        objects = self._get_minio_client().list_objects(
+            bucket_name=self.minio_bucket_name,
+            prefix=f"{str(self.minio_metric_datasets_dir / collection_name)}/",  # lists objects in "SOME_PATH/"
+        )
+        object_list = [obj.object_name for obj in objects]
+        object_sample_list = random.choices(object_list, k=k)
+        captions_cloud = [obj.split("/")[-1] for obj in object_sample_list]
+        imgs_cloud = [
+            Image.open(
+                self._get_minio_client().get_object(
+                    bucket_name=self.minio_bucket_name, object_name=str(obj)
+                )
+            )
+            for obj in object_sample_list
+        ]
+        return captions_cloud, imgs_cloud
+
+    def get_meta_json(self, model_name: str) -> dict[Union[list[int], str]]:
+        """
+        Get meta.json dictionary created during model training from cloud storage.
+        """
+        return json.load(
+            self._get_minio_client().get_object(
+                bucket_name=self.minio_bucket_name,
+                object_name=str(self.minio_models_dir / model_name / "meta.json"),
+            )
+        )
+
+    def get_weights_dict(self, model_name: str) -> dict[str]:
+        """
+        Get dictionary with directories to the embedder and trunk files for cloud storage.
+        """
+        return {
+            "trunk_minio": self.minio_models_dir / model_name / "trunk.pth",
```

Comment:
Because I need to pull those models locally from MinIO (so `minio` is like "from directory" and `local` is "to directory") to load them to `PyTorch`.

Replaced with Dataclass.